### PR TITLE
CTCTRADERS-108: XSD validation for ArrivalNotification XML 

### PR DESCRIPTION
**Pull request by MatthewAlexanderBeck**

### CODE REVIEW

File: app/utils/ValidateXML.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  private val schemaLang = javax.xml.XMLConstants.W3C_XML_SCHEMA_NS_URI
+
+  private def saxParser(schema: Schema): SAXParser = {
+    val saxParser: SAXParserFactory = javax.xml.parsers.SAXParserFactory.newInstance()
+    saxParser.setNamespaceAware(true)
+    saxParser.setSchema(schema)
+    saxParser.newSAXParser()
+  }
+
+  def validate(xml: String, fileKey: String): Try[String] = {
```
Comment (chris-ianson) [inline app/utils/ValidateXML.scala]: Maybe the fileKey could be an XSDFile trait which contains the full resource path. You could then have something like _ArrivalNotificationXSD_ which extends of this trait.

File: app/models/messages/XSDFile.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/*
```
Comment (chris-ianson) [inline app/models/messages/XSDFile.scala]: Do you think the XSDFile is in the right place? It's within messages directory but isn't a message.

File: app/utils/ValidateXML.scala [hunk 3]
```diff
@@ -0,0 +1,83 @@
+/*
```
Comment (chris-ianson) [inline app/utils/ValidateXML.scala]: Could the ValidateXML be made into a service?

File: test/models/messages/ArrivalNotificationSpec.scala [hunk 4]
```diff
@@ -432,5 +431,97 @@ class ArrivalNotificationSpec extends FreeSpec with MustMatchers
           Json.toJson(notification: ArrivalNotification) mustEqual json
       }
     }
+
+    "must convert to valid xml" in {
```
Comment (chris-ianson) [inline test/models/messages/ArrivalNotificationSpec.scala]: If we move the xsd into a service we could create another spec and move all validation checks into this

File: app/models/messages/XSDFile.scala (hunk 2 above)
Comment (MatthewAlexanderBeck) [reply app/models/messages/XSDFile.scala]: Agreed, moved outside of messages directory.

**Pull request merged**